### PR TITLE
Switch from pb alert feed to enhanced alert feed

### PR DESCRIPTION
**Pull request by lboyarsky**
Ticket: [Trip planner suggests the route affected by disruption](https://app.asana.com/0/810933294009540/1115762511783454)

The running theory is that erroneous behavior described in the ticket is caused by a race condition between two alert fetchers we have in place. 

This PR effectively reverts https://github.com/mbta/OpenTripPlanner/pull/1 and replaces existing  Protocol Buffers stream parsing with enhanced stream json parsing. The logic for applying alerts to trip plan remains the same as it was introduced in https://github.com/mbta/OpenTripPlanner/pull/1.

### CODE REVIEW

File: src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java [hunk 1]
```diff
@@ -90,7 +95,7 @@ protected void runPolling() {
                 throw new RuntimeException("Failed to get data from url " + url);
             }
 
-            final FeedMessage feed = FeedMessage.PARSER.parseFrom(data);
+            final FeedMessage feed = parseJson(IOUtils.toString(data));
```
Comment (paulswartz) [inline src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java]: I don't think we want to remove the existing PB support. Instead, we should additionally support fully parsing the enhanced JSON feed via the `real-time-alerts-enhanced` updater.

File: src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
     private String id;
 
-    private List<EnhancedAlert> enhancedAlerts = new ArrayList<>();
+    private List<String> activities = new ArrayList<>();
```
Comment (paulswartz) [inline src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java]: The default activities are `RIDE, BOARD, EXIT`.

File: src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java (hunk 1 above)
Comment (lboyarsky) [reply src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java]: If they have the same data (plus few extra things in enhanced json) and we're doing full parsing of json anyway, then why would we need to keep pb support in place?

File: src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java (hunk 2 above)
Comment (lboyarsky) [reply src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java]: Yes, those might be the defaults, but we're not reverting to the default values anywhere. We use whatever is in the json. If it's empty, then we're going to have the empty list, not the defaults.

File: src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java (hunk 1 above)
Comment (paulswartz) [reply src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java]: I'd like to be able to submit these changes upstream at some point, and I doubt they'll accept a PR which removes support for PB files :)

Comment (lboyarsky) [reply src/main/java/org/opentripplanner/updater/alerts/GtfsRealtimeAlertsUpdater.java]: Got it, good point, fixed.

File: src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java (hunk 2 above)
Comment (paulswartz) [reply src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java]: I was thinking more for the PB file which doesn't have these fields.

File: src/main/java/org/opentripplanner/updater/alerts/GtfsEnhancedRealtimeAlertsUpdater.java [hunk 3]
```diff
@@ -1,139 +1,246 @@
...
+            updaterManager.execute(new GraphWriterRunnable() {
+                @Override
+                public void run(Graph graph) {
+                    updateHandler.update(feed);
+                }
+            });
+
+            lastTimestamp = feedTimestamp;
+        } catch (Exception e) {
+            LOG.error("Error reading gtfs-realtime feed from " + url, e);
```
Comment (paulswartz) [inline src/main/java/org/opentripplanner/updater/alerts/GtfsEnhancedRealtimeAlertsUpdater.java]: should this (and other logging/strings) say realtime enhanced instead?

Comment (lboyarsky) [reply src/main/java/org/opentripplanner/updater/alerts/GtfsEnhancedRealtimeAlertsUpdater.java]: Fixed

File: src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java (hunk 2 above)
Comment (lboyarsky) [reply src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java]: Yep, fixed.

**Pull request merged**